ShellPkg/Library/UefiShellAcpiViewCommandLib/Parsers/Rimt/RimtParser.c

```diff
@@ -0,0 +1,876 @@
+/** @file
+  RISC-V IO Mapping Table (RIMT) parser
+
+  Copyright (c) 2025, Plasteli.net. All rights reserved.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+  Reference(s):
+    - ACPI 6.6 Specification
+    - RISC-V IO Mapping Table (RIMT) Specification Version v1.0, 2025-03-31: Ratified
+**/
+
+#include <IndustryStandard/RiscVIoMappingTable.h>
+#include <Library/PrintLib.h>
+#include <Library/UefiLib.h>
+#include "AcpiParser.h"
+#include "AcpiTableParser.h"
+
+typedef enum {
+  IOMMU = 0,
+  PCIERC,
+  PLATFORM,
+  UNSUPPORTED
+} RimtNodeTypes;
+
+// Local variables
+
+/**
+  This function validates RIMT Node type.
+
+  @param [in] Ptr     Pointer to the start of the field data.
+  @param [in] Length  Length of the field.
+  @param [in] Context Pointer to context specific information e.g. this
+                      could be a pointer to the ACPI table header.
+**/
+STATIC
+VOID
+EFIAPI
+RimtValidateNodeType (
+  IN UINT8   *Ptr,
+  IN UINT32  Length,
+  IN VOID    *Context
+  )
+{
+  UINT8  NodeType = *(UINT8 *)Ptr;
+
+  if (NodeType >= UNSUPPORTED) {
+    IncrementErrorCount ();
+    Print (L"\nERROR: Urecognized RIMT node type. Type value must be less than %d but got %d", UNSUPPORTED, NodeType);
+  }
+}
+
+/**
+  This function traces RIMT IOMMU Node Type field and presents in
+  human-readable form.
+
+  @param [in] Format  Optional format string for tracing the data.
+  @param [in] Ptr     Pointer to the start of the buffer.
+  @param [in] Length  Length of the field.
+**/
+STATIC
+VOID
+EFIAPI
+RimtNodeTypeToStr (
+  CONST CHAR16  *Format,
+  UINT8         *Ptr,
+  UINT32        Length
+  )
+{
+  CHAR8  *Str;
+  UINT8  NodeType;
+
+  NodeType = *(UINT8 *)Ptr;
+  if (NodeType == IOMMU) {

```

Comment:
I think it would be easier to use an array of strings, cf. the `ErstActionTable` for instance